src/controller/AutoCommissioner.cpp

```diff
@@ -249,56 +249,62 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
     if (err != CHIP_NO_ERROR)
     {
         ChipLogError(Controller, "Failed to perform commissioning step %d", static_cast<int>(report.stageCompleted));
-        return err;
     }
-    switch (report.stageCompleted)
+    else
     {
-    case CommissioningStage::kSendPAICertificateRequest:
-        SetPAI(report.Get<RequestedCertificate>().certificate);
-        break;
-    case CommissioningStage::kSendDACCertificateRequest:
-        SetDAC(report.Get<RequestedCertificate>().certificate);
-        break;
-    case CommissioningStage::kSendAttestationRequest:
-        // These don't need to be deep copied to local memory because they are used in this one step then never again.
-        mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements)
-            .SetAttestationSignature(report.Get<AttestationResponse>().signature);
-        // TODO: Does this need to be done at runtime? Seems like this could be done earlier and we woouldn't need to hold a
-        // reference to the operational credential delegate here
-        if (mOperationalCredentialsDelegate != nullptr)
+        switch (report.stageCompleted)
         {
-            MutableByteSpan nonce(mCSRNonce);
-            ReturnErrorOnFailure(mOperationalCredentialsDelegate->ObtainCsrNonce(nonce));
-            mParams.SetCSRNonce(ByteSpan(mCSRNonce, sizeof(mCSRNonce)));
+        case CommissioningStage::kSendPAICertificateRequest:
+            SetPAI(report.Get<RequestedCertificate>().certificate);
+            break;
+        case CommissioningStage::kSendDACCertificateRequest:
+            SetDAC(report.Get<RequestedCertificate>().certificate);
+            break;
+        case CommissioningStage::kSendAttestationRequest:
+            // These don't need to be deep copied to local memory because they are used in this one step then never again.
+            mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements)
+                .SetAttestationSignature(report.Get<AttestationResponse>().signature);
+            // TODO: Does this need to be done at runtime? Seems like this could be done earlier and we wouldn't need to hold a
+            // reference to the operational credential delegate here
+            if (mOperationalCredentialsDelegate != nullptr)
```

Comment:
Yes, it needs to be done at runtime

Reply:
okie dokie. Will remove the todo in a follow up (don't want to wait for the CI again)